dropzone/dropzone.d.ts

```diff
@@ -1,95 +1,128 @@
-    getUploadingFiles(): DropzoneFile[];
-
-    emit(eventName: string, file: DropzoneFile, str?: string);
-    emit(eventName: "thumbnail", file: DropzoneFile, path: string);
-    emit(eventName: "addedfile", file: DropzoneFile);
-    emit(eventName: "removedfile", file: DropzoneFile);
-    emit(eventName: "processing", file: DropzoneFile);
-    emit(eventName: "canceled", file: DropzoneFile);
-    emit(eventName: "complete", file: DropzoneFile);
-
-    emit(eventName: string, e: Event);
-    emit(eventName: "drop", e: Event);
-    emit(eventName: "dragstart", e: Event);
-    emit(eventName: "dragend", e: Event);
-    emit(eventName: "dragenter", e: Event);
-    emit(eventName: "dragover", e: Event);
-    emit(eventName: "dragleave", e: Event);
+	constructor( container: string, options?: DropzoneOptions );
+	constructor( container: HTMLElement, options?: DropzoneOptions );
+
+	static autoDiscover: boolean;
+	static options: any;
+	static confirm: ( question: string, accepted: () => void, rejected?: () => void ) => void;
+
+	files: DropzoneFile[];
+
+	enable(): void;
+	disable(): void;
+	destroy(): Dropzone;
+
+	on( eventName, callback: ( ...args ) => any );
+
+	off( eventName ): void;
+
+	addFile( file: DropzoneFile ): void;
+
+	removeFile( file: DropzoneFile ): void;
+
+	removeAllFiles( cancelIfNecessary?: boolean ): void;
+
+	processQueue(): void;
+
+	cancelUpload( file: DropzoneFile ): void;
+
+	processFiles ( files: DropzoneFile[] ): void;
+	processFile( file: DropzoneFile ): void;
+
+	uploadFile( file: DropzoneFile ): void;
+
+	getAcceptedFiles(): DropzoneFile[];
+
+	getRejectedFiles(): DropzoneFile[];
+
+	getQueuedFiles(): DropzoneFile[];
+
+	getUploadingFiles(): DropzoneFile[];
+
+	accept( file: DropzoneFile, done: ( error?: string ) => {} ):void;
+	getActiveFiles(): DropzoneFile[];
+	getFilesWithStatus( status: string ): DropzoneFile[];
+	enqueueFile( file: DropzoneFile ): void;
+	enqueueFiles( file: DropzoneFile[] ): void;
+	createThumbnail( file: DropzoneFile, callback?: (...any) => {}): any;
+	createThumbnailFromUrl( file: DropzoneFile, url: string, callback?: ( ...any ) => any ): any;
+
+	emit( eventName: string, file: DropzoneFile, str?: string );
+	emit( eventName: "thumbnail", file: DropzoneFile, path: string );
+	emit( eventName: "addedfile", file: DropzoneFile );
+	emit( eventName: "removedfile", file: DropzoneFile );
+	emit( eventName: "processing", file: DropzoneFile );
+	emit( eventName: "canceled", file: DropzoneFile );
+	emit( eventName: "complete", file: DropzoneFile );
+
+	emit( eventName: string, e: Event );
+	emit( eventName: "drop", e: Event );
+	emit( eventName: "dragstart", e: Event );
+	emit( eventName: "dragend", e: Event );
+	emit( eventName: "dragenter", e: Event );
+	emit( eventName: "dragover", e: Event );
+	emit( eventName: "dragleave", e: Event );
 }
 
 interface JQuery {
-    dropzone(options: DropzoneOptions): Dropzone;
+	dropzone( options: DropzoneOptions ): Dropzone;
+}
+
+declare module "Dropzone" {
```

Comment:
Technically per their documentation and source, they reference their objects with capital D. 

http://www.dropzonejs.com/#create-dropzones-programmatically